packages/button/src/Button.tsx

```diff
@@ -329,16 +106,76 @@ const Button: ExtendableBox<
     }: BaseButtonProps,
     ref: React.Ref<any>,
   ) => {
+    const colorSet = colorSets[variant];
+    const sizeSet = sizeSets[size];
+
     const { usingKeyboard: showFocus } = useUsingKeyboardContext();
 
     const commonProps = {
       ref,
       className: cx(
         baseStyle,
-        buttonSizes[size],
-        buttonVariants[variant],
+        css`
```

Comment:
I think @DesignerDave will suggest that we break this entire style up such that we do 

```
cx(baseStyle, colorSet[mode], ...)
```

Reply:
I'm open to exploring other patterns, but do think it's probably time for us to define one approach to this and be consistent, since right now we're doing it a bunch of different ways around the repo

Reply:
(ditto everywhere else)

Reply:
Could you go more into what about this approach has made it work the best? My main concerns with it are that it doesn't really enable DRY-ness and doesn't streamline the process of adding of new variants as much.